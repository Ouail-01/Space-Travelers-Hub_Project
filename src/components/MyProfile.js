import React from 'react';
import { useSelector } from 'react-redux';
import './MyProfile.css';

function MyProfile() {
  const missions = useSelector((state) => state.missions);
  const reservedMissions = missions.filter((item) => item.reserved === true);

  const rockets = useSelector((state) => state.rockets);
  const reservedRockets = rockets.filter((item) => item.reserved === true);

  if (reservedMissions.length === 0 && reservedRockets.length === 0) {
    return <div className="empty">No reserved Rockets/Missions</div>;
  }

  return (
    <div className="profileHandler">
      <div className="missions">
        <h2>My Missions</h2>
        <ul>
          {reservedMissions.map((item) => (
            <li key={item.id}>{item.name}</li>
          ))}
        </ul>
      </div>
      <div className="rockets">
        <h2>My Rockets</h2>
        <ul>
          {reservedRockets.map((rocket) => (
            <li key={rocket.id}>{rocket.name}</li>
          ))}
        </ul>
      </div>
    </div>
  );
}

export default MyProfile;
