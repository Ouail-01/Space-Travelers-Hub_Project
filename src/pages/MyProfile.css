.empty {
    display: flex;
    align-items: center;
    justify-content: center;
    height: 93vh;
    font-size: 40px;
    color: rgb(102, 102, 102);
}

.profileHandler {
    display: flex;
    width: 95%;
    margin: 50px auto;
}

.profileHandler ul {
    margin-top: 15px;
}

.profileHandler li {
    width: 95%;
    border: 2px solid rgb(135, 135, 135);
    padding: 35px 27px;
    list-style: none;
}

.missions {
    width: 50%;
}

.rockets {
    width: 50%;
}