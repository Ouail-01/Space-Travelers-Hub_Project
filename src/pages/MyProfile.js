import React from 'react';
import { useSelector } from 'react-redux';

function MyProfile() {
  const missions = useSelector((state) => state.missions);
  const reservedMissions = missions.filter((item) => item.reserved === true);

  const rockets = useSelector((state) => state.rockets);
  const reservedRockets = rockets.filter((item) => item.reserved === true);

  if (reservedMissions.length === 0 && reservedRockets.length === 0) {
    return <div>No reserved rockets/missions</div>;
  }

  return (
    <div>
      <div>
        <h2>My Missions</h2>
        <ul>
          {reservedMissions.map((item) => (
            <li key={item.id}>{item.name}</li>
          ))}
        </ul>
      </div>
      <div>
        <h2>My Rockets</h2>
        <ul>
          {reservedRockets.map((rocket) => (
            <li key={rocket.id}>{rocket.name}</li>
          ))}
        </ul>
      </div>
    </div>
  );
}

export default MyProfile;
