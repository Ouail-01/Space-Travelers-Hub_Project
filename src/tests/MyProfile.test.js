import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../Redux/configureStore';
import MyProfile from '../components/MyProfile';

const ProfileProvider = () => (
  <Provider store={store}>
    <MyProfile />
  </Provider>
);

describe('Test for the profile component', () => {
  it('Should be render profile correctly', () => {
    const myProfile = renderer.create(<ProfileProvider />);
    expect(myProfile).toMatchSnapshot();
  });
});
