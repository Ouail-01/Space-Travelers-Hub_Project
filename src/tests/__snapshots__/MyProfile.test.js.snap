// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for the profile component Should be render profile correctly 1`] = `
<div
  className="empty"
>
  No reserved Rockets/Missions
</div>
`;
